src/main/java/org/springframework/data/elasticsearch/core/document/MapDocument.java

```diff
@@ -114,6 +118,68 @@ public void setVersion(long version) {
 		this.version = version;
 	}
 
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.elasticsearch.core.document.Document#hasSeqNo()
+	 */
+	@Override
+	public boolean hasSeqNo() {
+		return this.seqNo != null && SequenceNumbers.isAssignedSeqNo(this.seqNo);
```

Comment:
```suggestion
		return this.seqNo != null;
```

I wouldn't do the validity check here but at the place where a `SeqNoPrimaryTerm`is constructed. Here in the `MapDocument` these are just numeric values coming from ES.

Reply:
Done